### PR TITLE
feat(cart): Add cart module package

### PR DESCRIPTION
**Pull request by olivermrbl**
**What**
Adds Cart Module package `@medusajs/cart`

Whether to stick with jest or replace with node's built-in testing tools (from v20) is to be discussed

Resolves CORE-1609

### CODE REVIEW

File: packages/cart/src/scripts/migration-down.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import { Modules } from "@medusajs/modules-sdk";
```
Comment (olivermrbl) [inline packages/cart/src/scripts/migration-down.ts:1]: comment: These scripts seem to contain a lot of duplicated code with little to no changes across modules. We should consider a refactor to eliminate the amount of boilerplate code

**Comment by olivermrbl** [comment]
I have left out the migration until I start integration testing

File: packages/cart/src/scripts/migration-down.ts (hunk 1 above)
Comment (adrien2p) [reply packages/cart/src/scripts/migration-down.ts:1]: I believe we can create utils function to build those script and export them from the modules

**Comment by olivermrbl** [comment]
@riqwan @pKorsholm, did any of you experience issues with building `medusa-react` after adding the new package? I find it strange that these changes affect that package. Am I missing something trivial?

**Review by adrien2p — APPROVED** [review]
LGTM, Maybe you can include a util proposal for the script in a later pr? or create a ticket that we can handle

**Comment by fPolic** [comment]
@olivermrbl I had the same build issue on my SC module PR.

**Comment by olivermrbl** [comment]
> I had the same build issue on my SC module PR

How did you resolve it?

**Comment by fPolic** [comment]
I [fixed](https://github.com/medusajs/medusa/pull/5923/commits/2b0d2857ca18db85f138b5044185c3fb0b407d96) the line it's complaining about but it is strange.

**Review by riqwan — APPROVED** [review]
LGTM! 🎉

**Comment by adrien2p** [comment]
> I have left out the migration until I start integration testing

Seems true, indeed the function does not return something all the time. In some cases it will return `Promise<unknown> | unknown` and on other cases it will return `void`. though, it should also fail on dev right?

**Comment by riqwan** [comment]
@olivermrbl should this require a changeset for the other packages thats affected here?

**Comment by olivermrbl** [comment]
> should this require a changeset for the other packages thats affected here?

Yes, nice catch. Will include it

**Review by fPolic — APPROVED** [review]
LGTM!

**Pull request merged**